src/iceberg/v2_metadata.cc

```diff
@@ -83,11 +83,18 @@ Status ManifestEntryAdapterV2::Init() {
 
   ICEBERG_ASSIGN_OR_RAISE(auto partition_type,
                           partition_spec_->PartitionType(*current_schema_));
+  partition_summary_ = std::make_unique<PartitionSummary>(*partition_type);
   manifest_schema_ = EntrySchema(std::move(partition_type));
   return ToArrowSchema(*manifest_schema_, &schema_);
 }
 
 Status ManifestEntryAdapterV2::Append(const ManifestEntry& entry) {
+  if (entry.IsAlive() && entry.sequence_number.has_value()) {
+    if (!min_sequence_number_.has_value() ||
```

Comment:
Why the maintenance of `min_sequence_number_` is duplicated in v2 and v3 but not v1? I think the best place is in the ManifestWriter::WriteEntry together with stats update.